client/src/components/App.js

```diff
@@ -14,16 +15,33 @@ import Dashboard from './Dashboard/Dashboard';
 import Personal from './Personal/Personal';
 
 class App extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            currentURL: "/"
+        }
+    }
     componentDidMount() {
         this.props.fetchUser();
+
+        // Listen to the URL for route changes and update the currentURL state accordingly
+        this.unlisten = history.listen((location) => {
+            this.setState({ currentURL: location.pathname });
+        });
+    }
+
+    componentWillUnmount() {
+        this.unlisten();
     }
 
     render() {
         return (
             <div className="container">
-                <BrowserRouter>
+                <Router history={history}>
                     <div>
-                        <Header />
+                        {/* Do not show the header when we're in the errors page */}
+                        {this.state.currentURL.includes('error') ? null : <Header />}
                         <Switch>
```

Comment:
Setting the state should cause a re-render...and thus, check against this each time.